File: app/src/main/java/com/thedesignerx/saim/new_message_app/ApiClient.java
package com.thedesignerx.saim.new_message_app;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import org.jetbrains.annotations.NotNull;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient {
    @NotNull
    private static Retrofit getRetrofit() {

        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        OkHttpClient client = new OkHttpClient.Builder().addInterceptor(interceptor).build();


        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("https://api.mocki.io/v1/")
                .client(client)
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        return retrofit;
    }
    public static ApiInterface getApiInterface (){
        ApiInterface apiInterface = getRetrofit().create(ApiInterface.class);
        return apiInterface;
    }

    public static ApiInterfaceRegister getApiInterfaceRegister() {
        ApiInterfaceRegister apiInterfaceRegister = getRetrofit().create(ApiInterfaceRegister.class);
        return apiInterfaceRegister;
    }
}

File: app/src/main/java/com/thedesignerx/saim/new_message_app/ApiClient.kt
package com.thedesignerx.saim.new_message_app

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ApiClient {
    private val retrofit: Retrofit
        private get() {
            val interceptor = HttpLoggingInterceptor()
            interceptor.setLevel(HttpLoggingInterceptor.Level.BODY)
            val client = OkHttpClient.Builder().addInterceptor(interceptor).build()
            return Retrofit.Builder()
                    .baseUrl("https://api.mocki.io/v1/")
                    .client(client)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
        }

    @JvmStatic
    val apiInterface: ApiInterface
        get() = retrofit.create(ApiInterface::class.java)

//    @JvmStatic
//    val apiInterfaceRegister: ApiInterfaceRegister
//        get() = retrofit.create(ApiInterfaceRegister::class.java)
}